sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -122,10 +125,22 @@ class HadoopTableReader(
     val attrsWithIndex = attributes.zipWithIndex
     val mutableRow = new SpecificInternalRow(attributes.map(_.dataType))
 
-    val deserializedHadoopRDD = hadoopRDD.mapPartitions { iter =>
+    val deserializedHadoopRDD = hadoopRDD.mapPartitionsWithIndex { (index, iter) =>
       val hconf = broadcastedHadoopConf.value.value
       val deserializer = deserializerClass.newInstance()
       deserializer.initialize(hconf, tableDesc.getProperties)
+      if (skipHeaderLineCount > 0 && isTextInputFormatTable) {
+        hadoopRDD.partitions(index) match {
+          case partition: HadoopPartition =>
+            if (partition.inputSplit.t.asInstanceOf[FileSplit].getStart() == 0) {
+              var i = 0
+              while (i < skipHeaderLineCount && iter.hasNext) {
+                i += 1
+                iter.next()
+              }
+            }
```

Comment:
There is an assumption here that 'skipHeaderLineCount' lines will be in the first split always ?

Reply:
@dongjoon-hyun I would like to use this PR, but it seems that in the current Spark version, performing an RDD transformation within another RDD transformation is not allowed, which results in an `RDDLacksSparkContextError()`. Do you know how to modify this?